pkg/sr/serde.go

```diff
@@ -258,28 +221,31 @@ func (s *Serde) Encode(v any) ([]byte, error) {
 	return s.AppendEncode(nil, v)
 }
 
-// AppendEncode appends an encoded value to b according to the schema registry
-// wire format and returns it. If EncodeFn was not used, this returns
-// ErrNotRegistered.
+// AppendEncode encodes a value and prepends the header according to the
+// configured SerdeHeader, appends it to b and returns b. If EncodeFn was not
+// registered, this returns ErrNotRegistered.
 func (s *Serde) AppendEncode(b []byte, v any) ([]byte, error) {
-	t, ok := s.loadTypes()[reflect.TypeOf(v)]
-	if !ok || (t.encode == nil && t.appendEncode == nil) {
-		return b, ErrNotRegistered
-	}
+	// Load tserde based on the registered type.
+	t := s.loadTypes()[reflect.TypeOf(v)]
 
-	b, err := s.header().AppendEncode(b, int(t.id), t.index)
```

Comment:
Looks like this was lost (and not added in one of the new top level functions)

Reply:
Oh I see, it's used in the new call to AppendEncode below.

Reply:
I might rather have a little bit of duplication, not sure yet.

Reply:
Yep leaning duplication; I rebased on master and squashed all of your commits and did a few touchups in #742.